.github/workflows/ci.yml

```diff
@@ -0,0 +1,23 @@
+name: Continuous Integration
+on:
+  workflow_dispatch: null
+  push:
+    branches:
+      - main
+  pull_request: null
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go-version: [ 'stable', 'oldstable' ]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-go@v4
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Continuous Integration
+        run: make ci
```

Comment:
Perhaps a bit late for this kind of feedback - but I'm kinda against this giant `make ci` step. This makes it needlessly difficult to figure out what's wrong when this fails. This bundles four distinct steps and one needs to go grep logs to understand what's up.

Fine to keep it, just my 2c.

Reply:
im not against, ill leave the ci step for engineers to run locally and ill spread out the fun into multiple steps in the action

Reply:
i was trying to resolve this one as i didnt think we had much more to talk about here as i had made the change as requested